### ISSUE TITLE
Use-cases are not properly cancelled on activity destroy events

### ISSUE THREAD

**Issue by MrHadiSatrio** [MrHadiSatrio/Journal3#13]
So far we've only been handling user-sourced cancellation events (through `BackButtonCancellationEventSource`). We should also channel such events on appropriate lifecycle state change (e.g., `Activity#onDestroy()`) lest we'd end up with leaks.

An implementation of it could look like so:

```kotlin
class LifecycleCancellationEventSource(
    private val lifecycle: Lifecycle
): EventSource {

    private val events: Flow<CancellationEvent> by lazy {
        callbackFlow {
            val observer = LifecycleEventObserver { source, event ->
                if (event != Lifecycle.Event.ON_DESTROY) return@LifecycleEventObserver
                trySend(CancellationEvent("system"))
            }
            lifecycle.addObserver(observer)
            awaitClose { lifecycle.removeObserver(observer) }
        }
    }

    constructor(activity: ComponentActivity) : this(activity.lifecycle)

    override fun events(): Flow<Event> {
        return events
    }
}
```

Post that, it's only a matter of ensuring all use-case declaration would have that as one of their event sources.

**Comment by MrHadiSatrio** [comment]
On second look, I'm puzzled as to why `CoroutineDispatchingUseCase` didn't do this job for us. It accepts a `CoroutineScope` and uses it to internally invoke `launch`. As `lifecycleScope` is used, logically, the decorated use-case should have been cancelled.

**Comment by MrHadiSatrio** [comment]
`CoroutineDispatchingUseCase` have been deprecated and is no longer being used as per changes merged with #14.
